org.lflang/src/org/lflang/generator/ts/TSReactionGenerator.kt

```diff
@@ -443,12 +502,222 @@ class TSReactionGenerator(
         )
     }
 
+    private fun generateSingleMutation(reactor : Reactor, mutation: Mutation): String {
```

Comment:
I think this is a great opportunity to refactor generateSingleReaction() function. Could you identify the common code lines between generateSingleReaction() and generateSingleMutation() separate the code lines out as functions, and have them called by generateSingleReaction() and generateSingleMutation()?